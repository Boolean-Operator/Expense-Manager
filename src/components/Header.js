import React from 'react';
import { NavLink } from 'react-router-dom';

// Header Component
const Header = () => (
  <header>
    <h1>Expensify Header</h1>
    <h2>
      <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard-*-</NavLink>
      <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    </h2>
  </header>
)
export default Header;