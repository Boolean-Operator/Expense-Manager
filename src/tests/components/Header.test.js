import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
  
  // expect(wrapper.find('h1').text()).toEqual('Expensify Header');

  // const renderer = new ReactShallowRenderer();
  // renderer.render(<Header />);
  // expect(renderer.getRenderOutput()).toMatchSnapshot();
  // console.log(renderer.getRenderOutput());
  
});