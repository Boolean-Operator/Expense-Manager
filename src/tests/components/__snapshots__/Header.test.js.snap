// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1
          title="Click to go to Main Dashboard"
        >
          Bean Counter
        </h1>
      </Link>
      <button
        className="button button-link"
        onClick={[Function]}
      >
        Log Out
      </button>
    </div>
  </div>
</header>
`;
